data/compute.go

```diff
@@ -177,3 +201,131 @@ func (a *API) GetTimeInRange(res http.ResponseWriter, req *http.Request) {
 	defer iter.Close(req.Context())
 	writeMongoJSONResponse(res, req, iter, logInfo)
 }
+
+func getLimitValueAsMmolL(p deviceParameter) (float64, error) {
+	limitValue, err := strconv.ParseFloat(p.Value, 64)
+	if err != nil {
+		return 0, err
+	}
+	if p.Unit == unitMgdL {
+		return limitValue / mgdlPerMmoll, nil
+	} else if p.Unit != unitMmolL {
+		return 0, errors.New("Invalid parameter unit")
+	}
+	return limitValue, nil
+}
+
+// Get the first / last data time for the specified user
+func (a *API) getDataSummaryRangeV1(ctx context.Context, traceID string, userID string, numDays int64) (*store.Date, string, error) {
+	timeIt(ctx, "dr") // data-range
+	defer timeEnd(ctx, "dr")
+
+	dates, err := a.store.GetDataRangeV1(ctx, traceID, userID)
```

Comment:
The start date of the range is useless in the further processings (and blip gets the range once you display patient data), you could reduce mongo request time if you just retrieved the end date.

Reply:
It's used by blip